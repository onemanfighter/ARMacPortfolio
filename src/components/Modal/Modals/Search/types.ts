export type SearchItem = {
  title: string;
  author: {
    firstName: string;
    lastName: string;
  };
};
