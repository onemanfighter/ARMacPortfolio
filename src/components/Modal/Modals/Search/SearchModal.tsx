import { useTranslation } from 'react-i18next';
import { SearchIcon } from '@assets';
import { Box, Input, InputGroup, InputLeftElement } from '@chakra-ui/react';
import { useEffect, useState } from 'react';
import { fuse } from '@providers';
import { settingsStore, darkModeColorSelector } from '@settingsStore';
import { useShallow } from 'zustand/react/shallow';

const SearchModal = () => {
  const { t } = useTranslation();
  const [searchText, setSearchText] = useState<string>('');
  const [searchResults, setSearchResults] = useState<any[]>([]);
  const { bottomBarBgColor } = settingsStore(useShallow(darkModeColorSelector));

  useEffect(() => {
    const timer = setTimeout(() => {
      if (searchText.length > 0) {
        const searchResults = fuse.search(searchText);
        setSearchResults(searchResults);
      }
    }, 300);

    return () => clearTimeout(timer);
  }, [searchText]);

  return (
    <Box
      width={'100%'}
      height={'100%'}
      bgColor={bottomBarBgColor}
      style={{ backdropFilter: 'blur(6px)' }}
      borderRadius={15}
      dropShadow={'md'}
      aria-label="search-modal"
    >
      <InputGroup>
        <InputLeftElement
          display={'flex'}
          flexDir={'column'}
          justifyContent={'center'}
          pointerEvents="none"
        >
          <SearchIcon color="gray" width={'1.5em'} height={'1.5em'} />
        </InputLeftElement>
        <Input
          aria-label="search-input"
          width={'100%'}
          borderRadius={15}
          height={12}
          type="text"
          borderBottom={1}
          borderBottomColor={'black'}
          borderColor={'black'}
          _hover={{ borderColor: 'black' }}
          _focus={{ borderColor: 'black' }}
          _placeholder={{ color: 'gray' }}
          placeholder={t('SearchModal.searchPlaceholder')}
          value={searchText}
          onChange={(e) => setSearchText(e.target.value)}
        />
      </InputGroup>
      {searchResults && (
        <Box aria-label="search-result">
          {searchResults.map((result) => (
            <Box key={result.item.title}>{result.item.title}</Box>
          ))}
        </Box>
      )}
    </Box>
  );
};

export default SearchModal;
