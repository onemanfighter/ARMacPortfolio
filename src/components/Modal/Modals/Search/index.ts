export { default as SearchModal } from './SearchModal';
export type { SearchItem } from './types';
