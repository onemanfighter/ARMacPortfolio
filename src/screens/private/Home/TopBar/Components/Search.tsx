import { SearchIcon } from '@assets';
import { Menu } from '@chakra-ui/react';
import { TopBarButton } from '@components';

const Search = () => {
  return (
    <Menu>
      <TopBarButton
        text=""
        onClick={() => {}}
        ariaLabel="search-top-bar-button"
        icon={<SearchIcon width="1.5em" height="1.5em" color="white" />}
      />
    </Menu>
  );
};

export default Search;
