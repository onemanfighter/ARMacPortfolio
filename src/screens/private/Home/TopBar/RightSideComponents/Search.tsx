import { SearchIcon } from '@assets';
import { IconButton } from '@chakra-ui/react';
import {
  darkModeColorSelector,
  settingsStore,
  useShallow,
} from '@settingsStore';
import { ModalID, modalSelector, uiStore } from '@uiStore';

const Search = () => {
  const { iconColor } = settingsStore(useShallow(darkModeColorSelector));
  const { openModal } = uiStore(useShallow(modalSelector));

  return (
    <IconButton
      onClick={() => {
        openModal(ModalID.SEARCH, () => {});
      }}
      height={'7'}
      aria-label="search-top-bar-button"
      icon={<SearchIcon width="1.2em" height="1.2em" color={iconColor} />}
      _hover={{ bg: 'transparent' }}
      _active={{ bg: '#f0f0f06f' }}
      variant={'ghost'}
    />
  );
};

export default Search;
