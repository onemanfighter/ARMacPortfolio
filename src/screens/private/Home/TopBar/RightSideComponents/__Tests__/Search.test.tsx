import { fireEvent, render, screen } from '@testing-library/react';
import Search from '../Search';

describe('Search', () => {
  it('should render', () => {
    const { container } = render(<Search />);

    expect(container).toMatchSnapshot();
  });

  it('should open pop-up when Search icon is tapped', () => {
    const { container } = render(<Search />);

    fireEvent.click(screen.getByLabelText('search-top-bar-button'));

    expect(container).toMatchSnapshot();
  });
});
