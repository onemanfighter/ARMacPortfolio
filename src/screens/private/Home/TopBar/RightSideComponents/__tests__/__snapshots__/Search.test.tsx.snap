// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should open pop-up when Search icon is tapped 1`] = `
<div>
  <button
    aria-label="search-top-bar-button"
    class="chakra-button css-jmqcxj"
    type="button"
  >
    <svg
      aria-hidden="true"
      color="#000000"
      fill="currentColor"
      focusable="false"
      height="1.2em"
      viewBox="0 0 24 24"
      width="1.2em"
    >
      <path
        d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
      />
    </svg>
  </button>
</div>
`;

exports[`Search should render 1`] = `
<div>
  <button
    aria-label="search-top-bar-button"
    class="chakra-button css-jmqcxj"
    type="button"
  >
    <svg
      aria-hidden="true"
      color="#000000"
      fill="currentColor"
      focusable="false"
      height="1.2em"
      viewBox="0 0 24 24"
      width="1.2em"
    >
      <path
        d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
      />
    </svg>
  </button>
</div>
`;
